Lombiq.Tests.UI/Services/WebDriverFactory.cs

```diff
@@ -128,37 +154,37 @@ private static TDriver SetCommonTimeouts<TDriver>(this TDriver driver, TimeSpan
         return driver;
     }
 
-    private static TDriver CreateDriver<TDriver>(Func<TDriver> driverFactory, IDriverConfig driverConfig)
-        where TDriver : RemoteWebDriver
+    private static async Task<TDriver> CreateDriverAsync<TDriver>(Func<TDriver> driverFactory, IDriverConfig driverConfig)
+        where TDriver : IWebDriver
     {
+        var version = "<UNKNOWN>";
         try
         {
+            version = await TryFindVersionAsync(driverConfig);
+
             // While SetUpDriver() does locking and caches the driver it's faster not to do any of that if the setup was
             // already done. For 100 such calls it's around 16 s vs <100 ms. The Lazy<T> trick taken from:
             // https://stackoverflow.com/a/31637510/220230
             _ = _driverSetups.GetOrAdd(driverConfig.GetName(), _ => new Lazy<bool>(() =>
             {
-                // Version selection based on the locally installed version is only available for Chrome, see:
-                // https://github.com/rosolko/WebDriverManager.Net/pull/91.
-                if (driverConfig is ChromeConfig)
-                {
-                    new DriverManager().SetUpDriver(driverConfig, VersionResolveStrategy.MatchingBrowser);
-                }
-                else
-                {
-                    new DriverManager().SetUpDriver(driverConfig);
-                }
```

Comment:
Why is this not necessary anymore?

Reply:
The version selection has been moved into the TryFindVersionAsync method so we always have a version parameter.